### ISSUE TITLE
AndInThreadsTest.java:49-51: Remove the use of the static...

### ISSUE THREAD

**Issue by 0pdd** [yegor256/cactoos#898]
The puzzle `829-59f865f8` from #829 has to be resolved:

https://github.com/yegor256/cactoos/blob/995eb06e36912fe63f3ab22d1f992284084ee698/src/test/java/org/cactoos/scalar/AndInThreadsTest.java#L49-L51

The puzzle was created by Paulo Benety on 22-May-18. 

Estimate: 30 minutes,     

If you have any technical questions, don't ask me, submit new tickets instead. The task will be "done" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

**Comment by 0crat** [comment]
@llorllale[/z](https://www.0crat.com/u/llorllale) please, pay attention to this issue

**Comment by 0crat** [comment]
Job [#898](https://github.com/yegor256/cactoos/issues/898) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
The job [#898](https://github.com/yegor256/cactoos/issues/898) assigned to @vssekorin[/z](https://www.0crat.com/u/vssekorin), here is [why](https://www.0crat.com/footprint/C63314D6Z/1851206082); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by 0crat** [comment]
The user @vssekorin[/z](https://www.0crat.com/u/vssekorin) resigned from [#898](https://github.com/yegor256/cactoos/issues/898), please stop working. Reason for job resignation: It is older than 10 days, see [§8](http://www.zerocracy.com/policy.html#8)

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#898` is not assigned, can't get performer

**Comment by 0crat** [comment]
Resigned on delay, see [§8](http://www.zerocracy.com/policy.html#8): -30 point(s) just awarded to @vssekorin[/z](https://www.0crat.com/u/vssekorin)

**Comment by 0crat** [comment]
The job [#898](https://github.com/yegor256/cactoos/issues/898) assigned to @iprogrammerr[/z](https://www.0crat.com/u/iprogrammerr), here is [why](https://www.0crat.com/footprint/C63314D6Z/f54f6876-7a17-4229-b25f-c085a38ae84d); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by Iprogrammerr** [comment]
@0crat waiting on outcome of #1008

**Comment by 0crat** [comment]
The job [#898](https://github.com/yegor256/cactoos/issues/898) assigned to @tomascury[/z](https://www.0crat.com/u/tomascury), here is [why](https://www.0crat.com/footprint/C63314D6Z/392c7a0c-88c5-4f39-8658-4f925507fc95); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by tomascury** [comment]
@0crat waiting on outcome of #947

**Comment by 0crat** [comment]
> @0crat waiting on outcome of #947 ([here](https://github.com/yegor256/cactoos/issues/898#issuecomment-461982385))

@tomascury The impediment for [#898](https://github.com/yegor256/cactoos/issues/898) was registered successfully by @tomascury[/z](https://www.0crat.com/u/tomascury)

**Comment by tomascury** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/cactoos/issues/898#issuecomment-466767690))

@tomascury The user @tomascury[/z](https://www.0crat.com/u/tomascury) resigned from [#898](https://github.com/yegor256/cactoos/issues/898), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @tomascury[/z](https://www.0crat.com/u/tomascury)

**Comment by 0crat** [comment]
The job [#898](https://github.com/yegor256/cactoos/issues/898) assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/cf3feb2c-4c1d-4fba-9750-3e4d5fb7d97c); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by victornoel** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/cactoos/issues/898#issuecomment-468939771))

@victornoel The user @victornoel[/z](https://www.0crat.com/u/victornoel) resigned from [#898](https://github.com/yegor256/cactoos/issues/898), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

**Comment by 0crat** [comment]
The job [#898](https://github.com/yegor256/cactoos/issues/898) assigned to @longstone[/z](https://www.0crat.com/u/longstone), here is [why](https://www.0crat.com/footprint/C63314D6Z/83d3cf06-d52f-4ddb-9e0f-e04a28a6e8e2); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#898` already assigned to @longstone, can't assign to @longstone

**Comment by longstone** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/cactoos/issues/898#issuecomment-471211703))

@longstone The user @longstone[/z](https://www.0crat.com/u/longstone) resigned from [#898](https://github.com/yegor256/cactoos/issues/898), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @longstone[/z](https://www.0crat.com/u/longstone)

**Comment by 0crat** [comment]
The job [#898](https://github.com/yegor256/cactoos/issues/898) assigned to @golyalpha[/z](https://www.0crat.com/u/golyalpha), here is [why](https://www.0crat.com/footprint/C63314D6Z/fb819772-b463-4913-8d2b-08ac341873a3); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by golyalpha** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/cactoos/issues/898#issuecomment-474549651))

@golyalpha The user @golyalpha[/z](https://www.0crat.com/u/golyalpha) resigned from [#898](https://github.com/yegor256/cactoos/issues/898), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @golyalpha[/z](https://www.0crat.com/u/golyalpha)

**Comment by 0crat** [comment]
The job [#898](https://github.com/yegor256/cactoos/issues/898) assigned to @atapin[/z](https://www.0crat.com/u/atapin), here is [why](https://www.0crat.com/footprint/C63314D6Z/c86da3f7-a16d-4506-808e-11a3f9bc1651); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by atapin** [comment]
@0crat waiting for #947

**Comment by 0crat** [comment]
> @0crat waiting for #947  ([here](https://github.com/yegor256/cactoos/issues/898#issuecomment-479179615))

@atapin The impediment for [#898](https://github.com/yegor256/cactoos/issues/898) was registered successfully by @atapin[/z](https://www.0crat.com/u/atapin)

**Comment by 0crat** [comment]
The job [#898](https://github.com/yegor256/cactoos/issues/898) assigned to @umbrah[/z](https://www.0crat.com/u/umbrah), here is [why](https://www.0crat.com/footprint/C63314D6Z/49f894c5-988e-4ff0-8c2f-06935422292d); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by Umbrah** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/cactoos/issues/898#issuecomment-487542907))

@Umbrah The user @umbrah[/z](https://www.0crat.com/u/umbrah) resigned from [#898](https://github.com/yegor256/cactoos/issues/898), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @umbrah[/z](https://www.0crat.com/u/umbrah)

**Comment by 0crat** [comment]
The job [#898](https://github.com/yegor256/cactoos/issues/898) assigned to @vzurauskas[/z](https://www.0crat.com/u/vzurauskas), here is [why](https://www.0crat.com/footprint/C63314D6Z/c5deef99-6426-4b1d-9142-3aba25dae845); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by vzurauskas** [comment]
@0crat refuse, this task depends on https://github.com/yegor256/cactoos/issues/947

**Comment by 0crat** [comment]
> @0crat refuse, this task depends on https://github.com/yegor256/cactoos/issues/947 ([here](https://github.com/yegor256/cactoos/issues/898#issuecomment-487966786))

@vzurauskas Can't understand "refuse,", try one of these:
  - `assign` Assign a performer to this issue
  - `boost` Set a boost for the job
  - `continue` Remove a job's impediment
  - `hello` Just say hello
  - `in` Register this issue in scope (WBS)
  - `out` Close the order and remove this job from scope
  - `pay` Pay a user some extra cash
  - `quality` Review a task
  - `resign` Resign from this issue
  - `status` Check the status of the job
  - `version` Print current version of the bot
  - `wait` Register an impediment for a job

**Comment by vzurauskas** [comment]
@0crat refuse this task, what's wrong with you? :)

**Comment by 0crat** [comment]
> @0crat refuse this task, what's wrong with you? :) ([here](https://github.com/yegor256/cactoos/issues/898#issuecomment-487968637))

@vzurauskas The user @vzurauskas[/z](https://www.0crat.com/u/vzurauskas) resigned from [#898](https://github.com/yegor256/cactoos/issues/898), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @vzurauskas[/z](https://www.0crat.com/u/vzurauskas)

**Comment by 0crat** [comment]
The job [#898](https://github.com/yegor256/cactoos/issues/898) assigned to @scristalli[/z](https://www.0crat.com/u/scristalli), here is [why](https://www.0crat.com/footprint/C63314D6Z/a570642a-50a5-4f3a-af8a-277bc35dd061); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by scristalli** [comment]
@0crat waiting for #947

**Comment by 0crat** [comment]
> @0crat waiting for #947  ([here](https://github.com/yegor256/cactoos/issues/898#issuecomment-494917957))

@scristalli The impediment for [#898](https://github.com/yegor256/cactoos/issues/898) was registered successfully by @scristalli[/z](https://www.0crat.com/u/scristalli)

**Comment by scristalli** [comment]
@llorllale this task requires a mutable collection by its description and previous comments, topic that is already present in #1008, addressed as a duplicate of #947, and postponed after release 1.0. Is the release expected soon?
Otherwise, is there any chance this task can be closed and put out of scope for now? It seems that it can't be solved at the moment and it only blocks slots for other jobs.

Thank you.

**Comment by scristalli** [comment]
@llorllale can I ask for an update on this? I'd just like to know what timeframe is expected for release 1.0, so that mutable collections can be discussed. I'd like not to refuse tasks, but depending on the actual feasability of this I don't want to hold it for months either.

**Comment by llorllale** [comment]
@scristalli just do this for now:

> Remove the use of the static method `Collections.synchronizedList`. Replace by an object-oriented approach.

**Comment by scristalli** [comment]
@llorllale I don't think the task can be split like so: an object-oriented approach would require a class like `Collections.SynchronizedList` (which is the one used by the static method), too bad it's private.
Unless of course leaving simple, unwrapped `ArrayList`, but that would make the code not thread-safe anymore.
The tests here create a collection, and then modify it by adding elements to it.

The only options I see:

1. Allow the introduction of mutable classes in Cactoos, so that we can use a mutable version of `Synced`. If this will be probably allowed after release 1.0, then I would like to know how far the project is from it.
2. Establish that the tests are wrong (because they go against the principles of EO), so what we need is to change the tests (admitting it is possible to do so and preserve their semantics).
3. Leave everything as it is, but this task would be unfeasible IMHO.

Have I missed something?

**Comment by llorllale** [comment]
@scristalli [you can proceed with `Synced`](https://github.com/yegor256/cactoos/issues/947#issuecomment-504246394)

**Comment by scristalli** [comment]
@llorllale thanks for unblocking this. I'm following discussion in #947, which I think needs to be solved before this. I would happily avoid to `Create a class similar to SyncCollection but mutable` if a more general solution is approved in the project.

**Comment by scristalli** [comment]
@0crat resign - this is taking way too long and I'd rather pass, in order to free up slots and improve reputation and metrics.

**Comment by 0crat** [comment]
> @0crat resign - this is taking way too long and I'd rather pass, in order to free up slots and im... ([here](https://github.com/yegor256/cactoos/issues/898#issuecomment-511935090))

@scristalli The user @scristalli[/z](https://www.0crat.com/u/scristalli) resigned from [#898](https://github.com/yegor256/cactoos/issues/898), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @scristalli[/z](https://www.0crat.com/u/scristalli)

**Comment by 0crat** [comment]
The job [#898](https://github.com/yegor256/cactoos/issues/898) assigned to @iakunin[/z](https://www.0crat.com/u/iakunin), here is [why](https://www.0crat.com/footprint/C63314D6Z/71e571fb-264b-4464-9278-a25710970e35); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by iakunin** [comment]
@paulodamaso, should we wait for #947 to be resolved?

**Comment by iakunin** [comment]
@paulodamaso, as I understand, I should create a mutable version of `org.cactoos.collection.Synced`, but it's not clear for me, how should I do that properly from architectural point of view.

Could you give me couple of advises?

**Comment by iakunin** [comment]
@0crat, waiting for @paulodamaso response

**Comment by 0crat** [comment]
Are you speaking to me or about me [here](https://github.com/yegor256/cactoos/issues/898#issuecomment-532338849); you must always start your message with my name if you want to address it to me, see [§1](http://www.zerocracy.com/policy.html#1)

**Comment by iakunin** [comment]
@0crat waiting for @paulodamaso response

**Comment by 0crat** [comment]
> @0crat waiting for @paulodamaso response ([here](https://github.com/yegor256/cactoos/issues/898#issuecomment-532339986))

@iakunin The impediment for [#898](https://github.com/yegor256/cactoos/issues/898) was registered successfully by @iakunin[/z](https://www.0crat.com/u/iakunin)

**Comment by fabriciofx** [comment]
@paulodamaso @scristalli @victornoel @iakunin @llorllale @vzurauskas @skapral I don't like the idea of mutable collections or even mutable classes in Cactoos. To me, it's thw wrong approach to this problem. My suggestion is to implement a [Persistent Data Structure](https://en.wikipedia.org/wiki/Persistent_data_structure) (there's a nice discussion about it [here](https://github.com/Kotlin/kotlinx.collections.immutable/issues/6)) to make our collections trully immutable. WDYT?

**Comment by vzurauskas** [comment]
@fabriciofx As far as I can tell, the [latest decision](https://github.com/yegor256/cactoos/issues/947#issuecomment-507053690) was for envelopes to delegate everything to the enveloped collection. If the collection is mutable, then the envelope is mutable; if the collection is immutable, then the envelope is immutable. For situations when client explicitly needs immutable collections, immutable decorators were supposed to be introduced.

I think this is a good decision because copying the entire collection every time something needs to be added or removed might be unfeasible sometimes, and also there are situations where it's very difficult to fit PDS into given architecture, and it's not obvious it would be worth it. Furthermore, I think there should be no logic in envelopes other than delegating to the enveloped object for the [reasons I outlined here](https://www.vzurauskas.com/2018/10/28/building-vertical-decorators-right/).

**Comment by victornoel** [comment]
@vzurauskas 
> …copying the entire collection every time something needs to be added or removed might be unfeasible sometimes.

This is not how persistent data structures work.

@fabriciofx That being said, maybe PDS is a good thing to implement, but it's another domain completely. Both needs exist (PDS or mutable collections) and they solve different problems. When doing OO, I personally see no problem with using mutable collections, except if it's used to implement DTO.

@paulodamaso Also another thing to take into account here is that it is great to want to reinvent the wheel, but we should be careful not to go too far where we are not able to succeed. Even implementing collections mutable or immutable is a big endeavour, and there exists already great OO libraries such as eclipse-collection (based on smalltalk collection design) and great PDS libraries in Java.

Personally I would continue the way it was planned, as @vzurauskas outlined.

**Comment by vzurauskas** [comment]
@victornoel Oh, right, my mistake.

**Comment by paulodamaso** [comment]
@fabriciofx @vzurauskas @victornoel @iakunin Here are my considerations about this ticket:
1. `cactoos` is, by definition, against the use of mutability (https://www.elegantobjects.org/#principles). So we shouldn't use mutable classes if we want to follow EO principles outlined in this link.
2. Following the extensive discussion in #898, #947 and #1008, the idea of providing the immutability by the use of decorators looks like reasonable to me, even though it goes against the principle outlined in 1
3. I like the idea of implementing PDS too, and I agree with @victornoel when he says that PDS is another problem domain; so, I think we should discuss it in another ticket.

So the real problem here is choosing between 1 and 2. I can't make this decision by myself since it changes the way cactoos follows EO principles. @yegor256 , what's your take on this? Which way should we follow, 1 (enforce immutability in all code blindly) or 2 (provide immutability by using decorators)?

**Comment by victornoel** [comment]
@paulodamaso this is one thing to have mutability in a class and have the "real world" state that an object represent to be mutable.

So a decorator for collections does not have to enforce immutability on the decorated collection, it should only prevent mutability of the reference to the decorated collection.

IMHO:
1. we shouldn't enforce immutability of decorated collection blindly: a decorator should do one and only one thing (so if it's job is mapping element of the decorated collection, it should only do that)
2. if we want a decorator that enforces immutability on the collection (an `ImmutableCollection` decorator), then it's a good idea, it can be composed with other when needed.

**Comment by fabriciofx** [comment]
@vzurauskas As @victornoel said, PDS are a completely different approach. It's made to perform immutable data sets properly.

@victornoel I see completly different. The immutable objects (mainly collections) is the heart of the problem because with mutable objects (or mutable collections) I can't share them properly. If I can shared, I've to copy them or I'll get problems with threads. I consider PDS into Cactoos a breakthrough in its development.

@paulodamaso I agree to discuss it in another ticket.

**Comment by skapral** [comment]
> @paulodamaso @scristalli @victornoel @iakunin @llorllale @vzurauskas @skapral I don't like the idea of mutable collections or even mutable classes in Cactoos. To me, it's thw wrong approach to this problem. My suggestion is to implement a Persistent Data Structure (there's a nice discussion about it here) to make our collections trully immutable. WDYT?

@fabriciofx I'll just answer that I am neutral to this. Generally, I have no certain favourite between mutable or persistent collections, both have a right to live.

**Comment by iakunin** [comment]
@paulodamaso so what’s your final decision? What should I do in this issue?

**Comment by paulodamaso** [comment]
@yegor256 Ping, I still want your opinion on https://github.com/yegor256/cactoos/issues/898#issuecomment-533117749 question, please

**Comment by iakunin** [comment]
@yegor256 sorry for being so impatient, could you give your opinion about question mentioned above?

**Comment by yegor256** [comment]
@paulodamaso I vote for decorators. Keep in mind that 100% immutability is not only impossible, but is against OOP spirit. Please, read this: https://www.yegor256.com/2016/09/07/gradients-of-immutability.html

**Comment by iakunin** [comment]
@paulodamaso is it ok that PR already has been merged, todo has disappeared from the code, but this task is still open?

**Comment by 0pdd** [comment]
The puzzle `829-59f865f8` has disappeared from the source code, that's why I closed this issue.

**Comment by 0pdd** [comment]
@0pdd 5 puzzles [#1217](https://github.com/yegor256/cactoos/issues/1217), [#1218](https://github.com/yegor256/cactoos/issues/1218), [#1219](https://github.com/yegor256/cactoos/issues/1219), [#1220](https://github.com/yegor256/cactoos/issues/1220), [#1221](https://github.com/yegor256/cactoos/issues/1221) are still not solved.

**Comment by 0crat** [comment]
The job [#898](https://github.com/yegor256/cactoos/issues/898) is now out of scope

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @iakunin[/z](https://www.0crat.com/u/iakunin), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +35 point(s) just awarded to @iakunin[/z](https://www.0crat.com/u/iakunin)

**Comment by 0crat** [comment]
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

**Comment by 0pdd** [comment]
@0pdd 5 puzzles [#1217](https://github.com/yegor256/cactoos/issues/1217), [#1218](https://github.com/yegor256/cactoos/issues/1218), [#1220](https://github.com/yegor256/cactoos/issues/1220), [#1221](https://github.com/yegor256/cactoos/issues/1221), [#1230](https://github.com/yegor256/cactoos/issues/1230) are still not solved; solved: [#1219](https://github.com/yegor256/cactoos/issues/1219).

**Comment by 0pdd** [comment]
@0pdd 5 puzzles [#1217](https://github.com/yegor256/cactoos/issues/1217), [#1218](https://github.com/yegor256/cactoos/issues/1218), [#1220](https://github.com/yegor256/cactoos/issues/1220), [#1221](https://github.com/yegor256/cactoos/issues/1221), [#1233](https://github.com/yegor256/cactoos/issues/1233) are still not solved; solved: [#1219](https://github.com/yegor256/cactoos/issues/1219), [#1230](https://github.com/yegor256/cactoos/issues/1230).

**Comment by 0pdd** [comment]
@0pdd 4 puzzles [#1217](https://github.com/yegor256/cactoos/issues/1217), [#1220](https://github.com/yegor256/cactoos/issues/1220), [#1221](https://github.com/yegor256/cactoos/issues/1221), [#1233](https://github.com/yegor256/cactoos/issues/1233) are still not solved; solved: [#1218](https://github.com/yegor256/cactoos/issues/1218), [#1219](https://github.com/yegor256/cactoos/issues/1219), [#1230](https://github.com/yegor256/cactoos/issues/1230).

**Comment by 0pdd** [comment]
@0pdd 4 puzzles [#1217](https://github.com/yegor256/cactoos/issues/1217), [#1220](https://github.com/yegor256/cactoos/issues/1220), [#1221](https://github.com/yegor256/cactoos/issues/1221), [#1291](https://github.com/yegor256/cactoos/issues/1291) are still not solved; solved: [#1218](https://github.com/yegor256/cactoos/issues/1218), [#1219](https://github.com/yegor256/cactoos/issues/1219), [#1230](https://github.com/yegor256/cactoos/issues/1230), [#1233](https://github.com/yegor256/cactoos/issues/1233).

**Comment by 0pdd** [comment]
@0pdd 3 puzzles [#1217](https://github.com/yegor256/cactoos/issues/1217), [#1220](https://github.com/yegor256/cactoos/issues/1220), [#1221](https://github.com/yegor256/cactoos/issues/1221) are still not solved; solved: [#1218](https://github.com/yegor256/cactoos/issues/1218), [#1219](https://github.com/yegor256/cactoos/issues/1219), [#1230](https://github.com/yegor256/cactoos/issues/1230), [#1233](https://github.com/yegor256/cactoos/issues/1233), [#1291](https://github.com/yegor256/cactoos/issues/1291).

**Comment by 0pdd** [comment]
@0pdd 2 puzzles [#1217](https://github.com/yegor256/cactoos/issues/1217), [#1221](https://github.com/yegor256/cactoos/issues/1221) are still not solved; solved: [#1218](https://github.com/yegor256/cactoos/issues/1218), [#1219](https://github.com/yegor256/cactoos/issues/1219), [#1220](https://github.com/yegor256/cactoos/issues/1220), [#1230](https://github.com/yegor256/cactoos/issues/1230), [#1233](https://github.com/yegor256/cactoos/issues/1233), [#1291](https://github.com/yegor256/cactoos/issues/1291).

**Comment by 0pdd** [comment]
@0pdd the puzzle [#1221](https://github.com/yegor256/cactoos/issues/1221) is still not solved; solved: [#1217](https://github.com/yegor256/cactoos/issues/1217), [#1218](https://github.com/yegor256/cactoos/issues/1218), [#1219](https://github.com/yegor256/cactoos/issues/1219), [#1220](https://github.com/yegor256/cactoos/issues/1220), [#1230](https://github.com/yegor256/cactoos/issues/1230), [#1233](https://github.com/yegor256/cactoos/issues/1233), [#1291](https://github.com/yegor256/cactoos/issues/1291).

**Comment by 0pdd** [comment]
@0pdd the puzzle [#unknown]() is still not solved.

**Comment by 0pdd** [comment]
@0pdd the puzzle [#1645](https://github.com/yegor256/cactoos/issues/1645) is still not solved; solved: [#unknown]().

**Comment by 0pdd** [comment]
@0pdd the puzzle [#unknown]() is still not solved; solved: [#1217](https://github.com/yegor256/cactoos/issues/1217), [#1218](https://github.com/yegor256/cactoos/issues/1218), [#1219](https://github.com/yegor256/cactoos/issues/1219), [#1220](https://github.com/yegor256/cactoos/issues/1220), [#1221](https://github.com/yegor256/cactoos/issues/1221), [#1230](https://github.com/yegor256/cactoos/issues/1230), [#1233](https://github.com/yegor256/cactoos/issues/1233), [#1291](https://github.com/yegor256/cactoos/issues/1291).

**Comment by 0pdd** [comment]
@0pdd the puzzle [#1730](https://github.com/yegor256/cactoos/issues/1730) is still not solved; solved: [#1217](https://github.com/yegor256/cactoos/issues/1217), [#1218](https://github.com/yegor256/cactoos/issues/1218), [#1219](https://github.com/yegor256/cactoos/issues/1219), [#1220](https://github.com/yegor256/cactoos/issues/1220), [#1221](https://github.com/yegor256/cactoos/issues/1221), [#1230](https://github.com/yegor256/cactoos/issues/1230), [#1233](https://github.com/yegor256/cactoos/issues/1233), [#1291](https://github.com/yegor256/cactoos/issues/1291).

**Comment by 0pdd** [comment]
@0pdd all 9 puzzles are solved here: [#1217](https://github.com/yegor256/cactoos/issues/1217), [#1218](https://github.com/yegor256/cactoos/issues/1218), [#1219](https://github.com/yegor256/cactoos/issues/1219), [#1220](https://github.com/yegor256/cactoos/issues/1220), [#1221](https://github.com/yegor256/cactoos/issues/1221), [#1230](https://github.com/yegor256/cactoos/issues/1230), [#1233](https://github.com/yegor256/cactoos/issues/1233), [#1291](https://github.com/yegor256/cactoos/issues/1291), [#1730](https://github.com/yegor256/cactoos/issues/1730).
